pkg/storage/stores/composite_store.go

```diff
@@ -28,7 +28,14 @@ type ChunkFetcherProvider interface {
 }
 
 type ChunkFetcher interface {
-	GetChunks(ctx context.Context, userID string, from, through model.Time, predicate chunk.Predicate) ([][]chunk.Chunk, []*fetcher.Fetcher, error)
+	GetChunks(
+		ctx context.Context,
+		userID string,
+		from,
+		through model.Time,
+		predicate chunk.Predicate,
+		storeChunksOverride *logproto.ChunkRefGroup,
```

Comment:
nit: I understand GetChunks does some extra filtering to the `storeChunksOverride` (validates and filters by the time range and fetcher). Still I think we should avoid calling `GetChunk` refs at all instead of passing the `storeChunksOverride` around. We may have a `GetFetcherForChunks` that returns grouped chunks by fetchers.